mslib/msui/mscolab.py

```diff
@@ -699,25 +698,24 @@ def after_login(self, emailid, url, r):
 
     def fetch_gravatar(self, refresh=False):
         # Display custom profile picture if exists
-        from mslib.mscolab.server import APP
-        with APP.app_context():
-            fm = FileManager(APP.config["MSCOLAB_DATA_DIR"])
-            img_data = fm.fetch_user_profile_image(self.user["id"])
-            if img_data:
-                img_byte_arr = io.BytesIO(img_data)
-                pixmap = QPixmap()
-                pixmap.loadFromData(img_byte_arr.getvalue())
-
-                if hasattr(self, 'profile_dialog'):
-                    self.profile_dialog.gravatarLabel.setPixmap(pixmap)
-
-                # set icon for user options toolbutton
-                icon = QtGui.QIcon()
-                icon.addPixmap(pixmap, QtGui.QIcon.Normal, QtGui.QIcon.Off)
-                self.ui.userOptionsTb.setIcon(icon)
-                return
+        url = urljoin(self.mscolab_server_url, 'fetch_profile_image')
+        data = {'user_id': str(self.user["id"])}
+        response = requests.get(url, data=data)
+        if response.status_code == 200:
+            img_data = response.content
+            img_byte_arr = io.BytesIO(img_data)
+            pixmap = QPixmap()
+            pixmap.loadFromData(img_byte_arr.getvalue())
```

Comment:
when someone wants to feed in here some none image data, does this throw an exception?